src/app/filter/filter.component.ts

```diff
@@ -0,0 +1,34 @@
+import { Component } from '@angular/core';
+import { FormControl, FormGroup } from '@angular/forms';
+import { Filters, FilterService } from './filter.service';
+
+@Component({
+  selector: 'app-filter',
+  templateUrl: './filter.component.html',
+  styleUrls: ['./filter.component.scss']
+})
+export class FilterComponent {
+  filterForm: FormGroup;
+
+  constructor(private readonly filterService: FilterService) {
+    const filterValues: Filters = this.filterService.values$.value;
+
+    this.filterForm = new FormGroup({
```

Comment:
use generic types for Forms

Reply:
moreover, you could do the same trick suggested below

Reply:
> use generic types for Forms

Can you explain this?

Reply:
```typescript
new FormGroup<myGenericType>({

})
```

FormGroup and FormControl on Angular14 supports generic type (`<T>`), you could create an interface for it, ex.

```typescript
interface LoginForm {
    email: FormControl<string>;
    password?: FormControl<string>;
}

const login = new FormGroup<LoginForm>({
    email: new FormControl<string>('', {nonNullable: true}),
    password: new FormControl<string>('', {nonNullable: true}),
});

login.removeControl('password');
```

source https://angular.io/guide/typed-forms#optional-controls-and-dynamic-groups

In the angular guide there isn't the `<string>` in FormControl but I think we could add it.